src/main.ts

```diff
@@ -39,6 +40,23 @@ const init = async (): Promise<void> => {
 
     const server = app.listen(config.port, () => logger.info(`Service listening on port ${config.port}`));
     server.on('close', async () => await knexConnection.destroy());
+    return server;
 };
 
-init();
+const main = async (): Promise<void> => {
+    const serverHandle = await init();
+
+    process.on('SIGINT', () => {
+        serverHandle.close(() => {
+            process.exit(0);
```

Comment:
I would expect process.exit(1). Is this expected?

Reply:
ok - on reading closer, we should not exit at all for SIGINT - as parent processes need to have a chance to tidy up too  - see https://www.cons.org/cracauer/sigint.html and if it were to exit it sound exit with "130" see http://www.tldp.org/LDP/abs/html/exitcodes.html